fl-server/src/serve_flists.rs

```diff
@@ -0,0 +1,214 @@
+use askama::Template;
+use axum::response::{Html, Response};
+use serde::Serialize;
+use std::path::PathBuf;
+use tokio::io;
+use tower::util::ServiceExt;
+use tower_http::services::ServeDir;
+use utoipa::ToSchema;
+
+use axum::{
+    body::Body,
+    http::{Request, StatusCode},
+    response::IntoResponse,
+};
+use axum_macros::debug_handler;
+use percent_encoding::percent_decode;
+
+#[debug_handler]
+pub async fn serve_flists(req: Request<Body>) -> impl IntoResponse {
```

Comment:
Can this be split into smaller functions, it's too nested it's really hard to follow and reason about the logic